### PR TITLE
support authenticating with a bearer token

### PR DESCRIPTION
**Pull request by pmenglund**

### CODE REVIEW

File: rockset_test.go [hunk 1]
```diff
@@ -119,6 +119,16 @@ func TestRockClient_withAPIServerEnv(t *testing.T) {
...
+func TestRockClient_withBearerToken(t *testing.T) {
+	token := "bearer token"
+	org := "rockset org"
+
+	rc, err := rockset.NewClient(rockset.WithBearerToken(token, org), rockset.WithAPIServer("apiserver"))
+	require.NoError(t, err)
+	assert.Equal(t, token, rc.Token)
+	assert.Equal(t, org, rc.Organization)
+}
+
```
Comment (mpatou) [inline rockset_test.go:131]: You most probably want to have a few more tests for the case where you have both the token and API key, and for the case when you have none.

**Review by mpatou — APPROVED** [review]
Approving but add the tests please. !
